crates/typst-cli/src/query.rs

```diff
@@ -0,0 +1,150 @@
+use codespan_reporting::diagnostic::{Diagnostic, Label};
+use codespan_reporting::term::{self, termcolor};
+use termcolor::{ColorChoice, StandardStream};
+use typst::diag::{bail, Severity, SourceDiagnostic, StrResult};
+use typst::doc::Document;
+use typst::eval::{eco_format, Tracer};
+use typst::World;
+use typst_library::prelude::EcoString;
+
+use crate::args::{CompileCommand, DiagnosticFormat, QueryCommand};
+use crate::world::SystemWorld;
+use crate::{color_stream, set_failed};
+
+/// Execute a compilation command.
+pub fn query(command: QueryCommand) -> StrResult<()> {
+    let mut world = SystemWorld::new(&CompileCommand {
+        // Little hack, only 3 fields are used
+        font_paths: command.font_paths.clone(),
+        input: command.input.clone(),
+        root: command.root.clone(),
+        output: None,
+        flamegraph: None,
+        open: None,
+        diagnostic_format: command.diagnostic_format,
+        ppi: 0.0,
+    })?;
+    tracing::info!("Starting querying");
+
+    let start = std::time::Instant::now();
+    // Reset everything and ensure that the main file is still present.
+    world.reset();
+    world.source(world.main()).map_err(|err| err.to_string())?;
+
+    let mut tracer = Tracer::default();
+
+    let result = typst::compile(&world, &mut tracer);
+    let duration = start.elapsed();
+
+    let warnings = tracer.warnings();
+
+    match result {
+        // Export the PDF / PNG.
+        Ok(document) => {
+            export(&document, &command)?;
```

Comment:
This is probably just copy-paste, but at least change the comment and maybe also change `export` to something more suitable.

Reply:
i'll have a look

Reply:
You're right, i had taken the whole function from compile-command as template. I don't say it's perfect. But it is working with my few examples and therefore a basis for discussion.